File: 20250506/linked_list.py
from node import Node

class LinkedList:
    "implementação de uma lista ligada"
    def __init__(self):
        self.head = None

    def prepend(self, data):
        novo_no = Node(data)
        novo_no.next = self.head
        self.head = novo_no
    
    def append(self, data):
        novo_no = Node(data)
        if not self.head:
            self.head = novo_no
            return
        atual =self.head
        while atual.next:
            atual = atual.next
        atual.next = novo_no

File: 20250508/LinkedList.java
public class LinkedList<T> {
    private Node<T> head; //referencia para o primeiro nó da lista

    public LinkedList(){
        this.head = null;
    }

    public void addFirst(T data){
        Node<T> novoNo = new Node<T>(data);
        novoNo.next = head;
        head = novoNo;
    }

    public void addLast(T data) {
        Node<T> novoNo = new Node<T>(data);
        if(head == null) {
            head = novoNo;
        } else {
            Node<T>atual = head;
            while(atual.next!=null){
                atual = atual.next;
            }
            atual.next=novoNo;
        }
    }

    public boolean remove(T data){
        if (head == null) {
            return false;
        }
        if(head == data){
            head = head.next;
            return true;
        }
        Node<T> atual=head;
        Node<T> anterior = null;
        while (atual !=null && atual.data != data){
            anterior = atual;
            atual=atual.next;
        }

        

        return false;
    }

}
